repobee_sanitizer/_sanitize_repo.py

```diff
@@ -102,6 +107,44 @@ def create_extension_command(self) -> plug.ExtensionCommand:
         )
 
 
+def _parse_file_list(file_list) -> List[pathlib.Path]:
+    return [
+        p.strip()
+        for p in file_list.read_text(encoding=_ASSUMED_ENCODING)
+        .strip()
+        .split("\n")
+    ]
+
+
+def _discover_dirty_files(repo_root) -> List[pathlib.Path]:
+    """
+    Returns:
+        A list of relative file paths for files that need to be sanitized.
+    """
+    return [
+        file
+        for file in list(repo_root.rglob("*"))
+        if ".git" not in str(file) and file.is_file() and _file_is_dirty(file)
+    ]
+
+
+def _file_is_dirty(file: pathlib.Path) -> bool:
+    """if a file has ANY block, pass it forward, its syntax will then be
+    checked when sanitized.
+    """
+
+    content = file.read_text(encoding=_ASSUMED_ENCODING).split("\n")
+    for line in content:
+        for block in [
+            "REPOBEE-SANITIZER-BLOCK",
+            "REPOBEE-SANITIZER-REPLACE-WITH",
+            "REPOBEE-SANITIZER-END",
+        ]:
+            if block in line:
+                return True
+    return False
```

Comment:
This function will need to be reconsidered, but let's go with this for a first implementation. It essentially has two problems:

1. Looping in Python is pretty slow, and so there could actually be a performance problem here (although unlikely). We may want to use regex instead, as its Python implementation is actually really fast (and in C).
2. It assumes that all files in the repository have the same encoding (here, UTF8). This is a, let's say, bold assumption. Text files may have other encodings, and binary blobs cannot be decoded as text at all (for example, an image).

But let's fix that some other time, I'll open an issue about it.